File: lwjgl3/src/main/java/io/github/msameer0/rhythmicrush/lwjgl3/atlases/PackPortalsAtlas.java
package io.github.msameer0.rhythmicrush.lwjgl3.atlases;

import com.badlogic.gdx.tools.texturepacker.TexturePacker;

public class PackPortalsAtlas {
    public static void main(String[] args) {
        TexturePacker.Settings settings = new TexturePacker.Settings();
        settings.maxWidth = 2048;
        settings.maxHeight = 2048;
        settings.edgePadding = true;
        settings.duplicatePadding = true;

        TexturePacker.process(settings,
            "assets/game/objects/portals",  // input
            "assets/game/objects",         // output
            "portals"                       // atlas name → spikes.atlas
        );
    }
}

File: lwjgl3/src/main/kotlin/io/github/msameer0/rhythmicrush/lwjgl3/atlases/PackPortalsAtlas.kt
package io.github.msameer0.rhythmicrush.lwjgl3.atlases

import com.badlogic.gdx.tools.texturepacker.TexturePacker

object PackPortalsAtlas {
    @JvmStatic
    fun main(args: Array<String>) {
        val settings = TexturePacker.Settings()
        settings.maxWidth = 2048
        settings.maxHeight = 2048
        settings.edgePadding = true
        settings.duplicatePadding = true

        TexturePacker.process(
            settings,
            "textures_to_put_in_atlases/objects/portals/back/",
            "assets/game/objects",
            "portals_back"
        )

        TexturePacker.process(
            settings,
            "textures_to_put_in_atlases/objects/portals/front/",
            "assets/game/objects",
            "portals_front"
        )
    }
}
